### PR TITLE
DSND-1902: Upgrade structured logging

### PR DESCRIPTION
**Pull request by johncook-ch**
* Added a `DataMapHolder` class to store a `DataMapHolder.Builder` instance that captures log data during message processing.
* Added a `String DataMapHolder.getRequestId()` to access the same request ID as used for logging
* Updated the `RequestLoggingInterceptor` to initialise the `DataMapHolder` with the request ID then ensures it is clean when processing is complete.
* Updated all logger calls to include the structured logging `DataMap` content.

The `RequestLoggingInterceptor` replaces the `StructuredLoggingFilter` described in the [Structured Logging](https://companieshouse.atlassian.net/wiki/spaces/TEAM4/pages/4253188169/Adding+Structured+Logging+to+Data+Sync+services) Confluence page. This page will be updated in due course.

[DSND-1902](https://companieshouse.atlassian.net/browse/DSND-1902)

[DSND-1902]: https://companieshouse.atlassian.net/browse/DSND-1902?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by kinpang-CH — APPROVED** [review]

**Review by sthompsonCH — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentService.java [hunk 1]
```diff
@@ -171,26 +171,24 @@ public void patchCompanyNameStatus(String companyNumber, String companyName,
         } catch (DataAccessException ex) {
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentService.java:171]: Line 169 has a debug log with no logMap

File: src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentService.java [hunk 2]
```diff
@@ -205,7 +203,7 @@ public void patchNewAppointmentCompanyNameStatus(String companyNumber, String ap
...
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentService.java:203]: Line 201 debug log with no logMap

File: src/main/java/uk/gov/companieshouse/company_appointments/api/CompanyMetricsApiService.java [hunk 3]
```diff
@@ -30,6 +31,7 @@ public CompanyMetricsApiService(@Value("${company-metrics-api.endpoint}") String
     public ApiResponse<MetricsApi> invokeGetMetricsApi(String companyNumber)
             throws ServiceUnavailableException, NotFoundException {
         InternalApiClient internalApiClient = apiClientService.getInternalApiClient();
+        internalApiClient.getHttpClient().setRequestId(DataMapHolder.getRequestId());
         internalApiClient.setBasePath(metricsUrl);
 
         PrivateCompanyMetricsGet companyMetricsGet =
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/company_appointments/api/CompanyMetricsApiService.java:37]: Log messages in this class need a logMap

**Review by cgregoryCH — CHANGES_REQUESTED** [review]
ResourceChangedApiService, needs a request ID on its api client as well as logMaps in its log statements.

**Review by cgregoryCH — APPROVED** [review]

**Pull request merged**